src/components/Autosuggest/index.js

```diff
@@ -1,96 +1,86 @@
-    this.props.onSearch(value);
-    this.setState({ userInput: value });
-  };
+  const onSuggestionsFetchRequested = useCallback(
+    ({ value }) => {
+      onSearch(value);
+      setUserInput(value);
+    },
+    [onSearch],
+  );
 
-  onSuggestionsClearRequested = () => {
-    this.setState({ suggestions: [] });
-  };
+  const toggleFocus = useCallback(() => {
+    setIsFocused((value) => !value);
+  }, []);
 
-  toggleFocus = () => {
-    this.setState({ isFocused: !this.state.isFocused });
-  };
+  const onSuggestionSelect = useCallback(
+    (event, { suggestion }) => {
+      event.preventDefault();
+      onSelect(suggestion.id);
+    },
+    [onSelect],
+  );
 
-  onSuggestionSelect = (e, { suggestion }) => {
-    e.preventDefault();
-    this.props.onSelect(suggestion.id);
+  const inputProps = {
+    value: userInput,
+    onBlur: toggleFocus,
+    onChange: onChange,
+    onFocus: toggleFocus,
   };
 
-  render() {
-    const { isFocused, userInput } = this.state;
-    const {
-      label,
-      suggestions,
-      getSuggestionValue,
-      renderSuggestion,
-    } = this.props;
-
-    const inputProps = {
-      value: userInput,
-      onBlur: this.toggleFocus,
-      onChange: this.onChange,
-      onFocus: this.toggleFocus,
-    };
-
-    return (
-      <div css={[styles.root, isFocused && styles.active]}>
-        <label css={styles.container}>
-          {label && (
-            <div
-              css={[
-                styles.label,
-                (isFocused || userInput) && styles.activeLabel,
-              ]}
-            >
-              {label}
-            </div>
-          )}
-        </label>
-        <Autosuggest
-          suggestions={suggestions}
-          onSuggestionsFetchRequested={this.onSuggestionsFetchRequested}
-          onSuggestionsClearRequested={this.onSuggestionsClearRequested}
-          getSuggestionValue={getSuggestionValue}
-          onSuggestionSelected={this.onSuggestionSelect}
-          renderSuggestion={renderSuggestion}
-          inputProps={inputProps}
-        />
-      </div>
-    );
-  }
+  return (
+    <div css={[styles.root, isFocused && styles.active]}>
+      <label css={styles.container}>
+        {label && (
+          <div
+            css={[styles.label, (isFocused || userInput) && styles.activeLabel]}
+          >
+            {label}
+          </div>
+        )}
+      </label>
+      <ReactAutosuggest
+        suggestions={suggestions}
+        onSuggestionsFetchRequested={onSuggestionsFetchRequested}
+        getSuggestionValue={getSuggestionValue}
+        onSuggestionSelected={onSuggestionSelect}
+        renderSuggestion={renderSuggestion}
+        inputProps={inputProps}
+      />
+    </div>
+  );
 }
 
-export default Typeahead;
+Autosuggest.propTypes = {
+  getSuggestionValue: PropTypes.func.isRequired,
+  onSearch: PropTypes.func.isRequired,
+  renderSuggestion: PropTypes.func.isRequired,
+  suggestions: PropTypes.array.isRequired,
+};
```

Comment:
I haven't seen a propType in awhile 😄. Are we thinking this will be a low level placeholder for Typescript and is something we want to use as much as possible?

Reply:
I'd rather add TypeScript in a transitional manner (allowing js).